src/core/desktopentry.cpp

```diff
@@ -183,20 +188,55 @@ void DesktopEntry::parseEntry(const QString& text) {
 	}
 
 	finishCategory();
+	return data;
+}
+
+void DesktopEntry::updateState(const ParsedDesktopEntryData& newState) {
+	this->bName = newState.name;
+	this->bGenericName = newState.genericName;
+	this->bStartupClass = newState.startupClass;
+	this->bNoDisplay = newState.noDisplay;
+	this->bComment = newState.comment;
+	this->bIcon = newState.icon;
+	this->bExecString = newState.execString;
+	this->bCommand = newState.command;
+	this->bWorkingDirectory = newState.workingDirectory;
+	this->bRunInTerminal = newState.terminal;
+	this->bCategories = newState.categories;
+	this->bKeywords = newState.keywords;
+
+	this->state = newState;
+	this->updateActions(newState.actions);
+}
+
+void DesktopEntry::updateActions(const QHash<QString, DesktopActionData>& newActions) {
+	for (auto* action: this->mActions.values()) {
+		action->deleteLater();
+	}
+	this->mActions.clear();
+
+	for (const auto& [actionName, actionData]: newActions.asKeyValueRange()) {
+		auto* action = new DesktopAction(actionData.id, this);
+		action->mEntries = actionData.entries;
+		action->mName = actionData.name;
+		action->mIcon = actionData.icon;
+		action->mExecString = actionData.execString;
+		action->mCommand = actionData.command;
+		this->mActions.insert(actionName, action);
+	}
```

Comment:
This should be handled with a diff update, and bindables in DesktopAction should be used.